clean/post_officer_history.py

```diff
@@ -0,0 +1,255 @@
+
+
+def split_names(df):
+    names = (
+        df.officer_name.str.lower()
+        .str.strip()
+        .str.extract(r"(\w+(?:'\w+)?),? (\w+)(?: (\w+))?")
+    )
+
+    df.loc[:, "last_name"] = names[0]
+    df.loc[:, "first_name"] = names[1]
+    df.loc[:, "middle_name"] = names[2]
+    return df.drop(columns=["officer_name"])
+
+
+def clean_post_agency_row(df):
+    for col in df.columns:
+        if col.startswith("agency"):
+            df[col] = (
+                df[col]
+                .str.strip()
+                .str.lower()
+                .str.replace(r"(\w+) = (\w+)", r"\1 \2", regex=True)
+                .str.replace(
+                    r"^orleans parish coroner\'s office",
+                    "orleans coroners office",
+                    regex=True,
+                )
+                .str.replace(r"(\w+) Ã¢â‚¬â€ (\w+)", r"\1 \2", regex=True)
+                .str.replace(r"1st parish court", "first court", regex=False)
+                .str.replace(
+                    r"(\w+) Ã‚Â© (\w+)\/(\w+)\/(\w+)", r"\1 \2/\3/\4", regex=True
+                )
+                .str.replace(r" Â© ", "", regex=True)
+                .str.replace(r"--BULL-TIME", "FULL-TIME", regex=True)
+                .str.replace(r"_(\w+)\/(\w+)\/(\w+)â€”_", r"\1/\2/\3", regex=True)
+                .str.replace(r" â€”= ", "", regex=True)
+                .str.replace(r" +", " ", regex=True)
+                .str.replace(r" & ", "", regex=True)
+                .str.replace(r" - ", "", regex=True)
+                .str.replace(r" _?â€\” ", "", regex=True)
+                .str.replace(r" _ ", "", regex=True)
+                .str.replace(r" = ", "", regex=True)
+                .str.replace(r"^ (\w+)", r"\1", regex=True)
+                .str.replace(r"^st (\w+)", r"st\1", regex=True)
+                .str.replace(r" of ", "", regex=True)
+                .str.replace(r" p\.d\.", " pd", regex=True)
+                .str.replace(r" pari?s?h? so", " so", regex=True)
+                .str.replace(r" Â§ ", "", regex=True)
+                .str.replace(r" â€˜", "", regex=True)
+                .str.replace(r"(\.|\,)", "", regex=True)
+                .str.replace(r"miss river", "river", regex=False)
+                .str.replace(r" ~ ", "", regex=False)
+                .str.replace(r"(\w+) _ (\w+)", r"\1 \2", regex=True)
+                .str.replace("new orleans harbor", "orleans harbor", regex=False)
+                .str.replace(
+                    "probation & parcole - adult",
+                    "adult probation",
+                    regex=True,
+                )
+            )
+    return df
+
+
+def extract_agency(df):
+    for col in df.columns:
+        if col.startswith("agency"):
+            df[col] = (
+                df[col]
+                .fillna("")
+                .str.lower()
+                .str.strip()
+                .str.replace(r"(\w{1}? ?\w+ ?\w+? ?\w+) (.+)", r"\1", regex=True)
```

Comment:
`{1}` is completely redundant. `\w` always matches exactly one letter .

Reply:
Thanks